pubs/plugs/git/git.py

```diff
@@ -0,0 +1,107 @@
+import os
+import sys
+import argparse
+from subprocess import Popen, PIPE
+from pipes import quote as shell_quote
+
+from ...plugins import PapersPlugin
+from ...events import PaperChangeEvent, PostCommandEvent
+
+
+GITIGNORE = """# files or directories for the git plugin to ignore
+.cache/
+"""
+
+
+class GitPlugin(PapersPlugin):
+    """The git plugin creates a git repository in the pubs directory and commit the changes
+    to the pubs repository everytime a paper is modified.
+
+    It also add the `pubs git` subcommand, so git commands can be executed in the git repository
+    from the command line.
+    """
+
+    name = 'git'
+    description = "Run git commands in the pubs directory"
+
+    def __init__(self, conf, ui):
+        self.ui = ui
+        self.pubsdir = conf['main']['pubsdir']
+        self.manual  = conf['plugins'].get('git', {}).get('manual', False)
+        self.quiet   = conf['plugins'].get('git', {}).get('quiet', True)
+        self.list_of_changes = []
+        self._gitinit()
+
+    def _gitinit(self):
+        """Initialize the git repository if necessary."""
+        # check that a `.git` directory is present in the pubs dir
+        git_path = os.path.expanduser(os.path.join(self.pubsdir, '.git'))
+        if not os.path.isdir(git_path):
+            self.shell('init')
+        # check that a `.gitignore` file is present
+        gitignore_path = os.path.expanduser(os.path.join(self.pubsdir, '.gitignore'))
+        if not os.path.isfile(gitignore_path):
+            print('bla')
+            with open(gitignore_path, 'w') as fd:
+                fd.write(GITIGNORE)
+
+    def update_parser(self, subparsers, conf):
+        """Allow the usage of the pubs git subcommand"""
+        git_parser = subparsers.add_parser(self.name, help=self.description)
+        # FIXME: there may be some problems here with the -c argument being ambiguous between
```

Comment:
> If I am correct everything after the pubs git is going to be passed to git.

I don't believe that is the case at the moment.

```
pubs git -c color.ui=always log
error: No configuration found at path color.ui=always. Maybe you need to initialize your repository with `pubs init` or specify a --config argument.
```
If I'm understanding this correctly, then this is the sort of syntax we want right?
```
pubs $PUBS_ARGS git $GIT_ARGS
```

Reply:
Yes, basically, we do a parse_know_args pass for the `--force-colors` and `-c` arguments, and because the `-c` argument would modify whether we have a git plugin or not, it is quite iffy to do anything about it. There are ways to consider where arguments are, but they are a bit hacky. The problem is of course that `-c` could be quite useful for git commands, but it would affect a minority of users of a small pool of users using the git shortcut of small pool of users using the git plugin... my opinion is not to address it, with a workaround a simple as going to the pubs folder and typing git command directly there.

Reply:
Yeah that makes sense!